pkgs/cli/src/cli_integration_tests.zig

```diff
@@ -0,0 +1,257 @@
+        return allocator.dupe(u8, response_buffer[0..bytes_read]);
+    }
+
+    /// Wait for server to be ready
+    fn waitForServer(_: std.mem.Allocator, host: []const u8, port: u16, timeout_ms: u32) !void {
+        const start_time = std.time.milliTimestamp();
+
+        while (std.time.milliTimestamp() - start_time < timeout_ms) {
+            const address = net.Address.parseIp4(host, port) catch {
+                std.time.sleep(100 * std.time.ns_per_ms);
+                continue;
+            };
+
+            var connection = net.tcpConnectToAddress(address) catch {
+                std.time.sleep(100 * std.time.ns_per_ms);
+                continue;
+            };
+            connection.close();
+            return;
+        }
+
+        return error.ServerNotReady;
+    }
+};
+
+test "CLI sim command starts metrics server" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    // Test metrics initialization (simulating what CLI does)
+    try metrics.init(allocator);
+
+    // Test that metrics can be written to a buffer
+    var metrics_output = std.ArrayList(u8).init(allocator);
+    defer metrics_output.deinit();
+
+    // This should not fail
+    metrics.writeMetrics(metrics_output.writer()) catch |err| {
+        // If metrics writing fails, it's not a critical error for this test
+        std.debug.print("Metrics writing failed: {}\n", .{err});
+    };
+
+    // Test port availability (simulating metrics server port selection)
+    const test_port = try TestUtils.findAvailablePort(allocator);
+    try std.testing.expect(test_port >= 9668 and test_port <= 9672);
+
+    // Test that we can create a simple HTTP server context (without starting it)
+    const address = try net.Address.parseIp4("127.0.0.1", test_port);
+    var server = try address.listen(.{});
+    defer server.deinit();
+}
+
+test "CLI sim command with custom port" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    // Test with different port
+    const custom_port = try TestUtils.findAvailablePort(allocator);
+    try std.testing.expect(custom_port >= 9668 and custom_port <= 9672);
+
+    // Test metrics initialization
+    try metrics.init(allocator);
+
+    // Test that we can create a server on custom port
+    const address = try net.Address.parseIp4("127.0.0.1", custom_port);
+    var server = try address.listen(.{});
+    defer server.deinit();
+}
+
+test "CLI sim command argument validation" {
+
+    // Test valid sim command arguments
+    const valid_args = [_][]const u8{ "sim", "--metrics-port", "9668" };
+    try std.testing.expect(valid_args.len == 3);
+    try std.testing.expect(std.mem.eql(u8, valid_args[0], "sim"));
+    try std.testing.expect(std.mem.eql(u8, valid_args[1], "--metrics-port"));
+    try std.testing.expect(std.mem.eql(u8, valid_args[2], "9668"));
+
```

Comment:
i don't think there is any value in just matching the string that you yourself allocated in array, remove this test unless this is to actually run a cli with these args and test that started node behaves as expected